ibm/service/catalogmanagement/resource_ibm_cm_offering.go

```diff
@@ -179,140 +2460,3603 @@ func resourceIBMCmOfferingCreate(d *schema.ResourceData, meta interface{}) error
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			media = append(media, mediaItemMap)
+		}
+	}
+	if err = d.Set("media", media); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting media: %s", err))
+	}
+	if offering.DeprecatePending != nil {
+		deprecatePendingMap, err := resourceIBMCmOfferingDeprecatePendingToMap(offering.DeprecatePending)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if err = d.Set("deprecate_pending", []map[string]interface{}{deprecatePendingMap}); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting deprecate_pending: %s", err))
+		}
+	}
+	if err = d.Set("product_kind", offering.ProductKind); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting product_kind: %s", err))
+	}
+	badges := []map[string]interface{}{}
+	if offering.Badges != nil {
+		for _, badgesItem := range offering.Badges {
+			badgesItemMap, err := resourceIBMCmOfferingBadgeToMap(&badgesItem)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			badges = append(badges, badgesItemMap)
+		}
+	}
+	if err = d.Set("badges", badges); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting badges: %s", err))
+	}
+	if err = d.Set("rev", offering.Rev); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting rev: %s", err))
+	}
+	if err = d.Set("offering_id", offering.ID); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting offering_id: %s", err))
+	}
 
 	return nil
 }
 
-func resourceIBMCmOfferingRepoInfoToMap(repoInfo catalogmanagementv1.RepoInfo) map[string]interface{} {
-	repoInfoMap := map[string]interface{}{}
+func resourceIBMCmOfferingUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	catalogManagementClient, err := meta.(conns.ClientSession).CatalogManagementV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updateOfferingOptions := &catalogmanagementv1.UpdateOfferingOptions{}
+
+	getOfferingOptions := &catalogmanagementv1.GetOfferingOptions{}
+	parts, err := flex.SepIdParts(d.Id(), "/")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getOfferingOptions.SetCatalogIdentifier(parts[0])
+	getOfferingOptions.SetOfferingID(parts[1])
+	offering, response, err := catalogManagementClient.GetOfferingWithContext(context, getOfferingOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] GetOfferingWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("GetOfferingWithContext failed %s\n%s", err, response))
+	}
+
+	updateOfferingOptions.SetCatalogIdentifier(*offering.CatalogID)
+	updateOfferingOptions.SetOfferingID(*offering.ID)
+	ifMatch := fmt.Sprintf("\"%s\"", *offering.Rev)
+	updateOfferingOptions.IfMatch = &ifMatch
+
+	hasChange := false
+
+	if d.HasChange("catalog_identifier") {
+		return diag.FromErr(fmt.Errorf("Cannot update resource property \"%s\" with the ForceNew annotation."+
```

Comment:
You can handle this condition just by setting ForceNew : true on that particular argument